src/optimizer/optimizer.cpp

```diff
@@ -68,6 +71,20 @@ void Optimizer::OptimizeLoop(int root_group_id,
       std::unique_ptr<OptimizerTaskStack>(new OptimizerTaskStack());
   metadata_.SetTaskPool(task_stack.get());
 
+  if (root_context->metadata->enable_transitive_predicates_) {
+    // Apply query rewrite rules
+    task_stack->Push(new TopDownRewrite(root_group_id, root_context,
+					RewriteRuleSetName::TRANSITIVE_PREDICATES));
+
+    ExecuteTaskStack(*task_stack, root_group_id, root_context);
+  }
+
+  // Apply query rewrite rules
+  task_stack->Push(new TopDownRewrite(root_group_id, root_context,
+                                      RewriteRuleSetName::SIMPLIFY_PREDICATES));
```

Comment:
Why is simplifying predicates executed after transitive predicates? Will this not apply the transitive predicate rule on trivial predicates that will just get thrown out by the simplify predicate rule? Seems like we can save some work by flipping the order of these